File: C/Armstrong.c
#include<stdio.h>
#include<math.h>
int main(){
	int num,count=0,arm=0,val,rem;
	scanf("%d",&num);
	val=num;
	while(num>0){
		num/=10;
		count++;
	}
	num=val;
	while(num>0){
		rem=num%10;
		num/=10;
		arm=arm+pow(rem,count);
	}
	if(val==arm)
		printf("Armstrong Number");
	else
		printf("Not an Armstrong Number");
	return 0;
}

File: Java/armstrong.java
import java.util.Scanner;

public class ArmstrongNum {

    public static void main(String[] args) {

        int originalNum, digit, cubeSum = 0,num;
        
        System.out.println("Enter the number:");
        Scanner sc = new Scanner(System. in );
        num = sc.nextInt();
        originalNum = num;

        while (num!= 0)
        {
            digit = num % 10;
            cubeSum += Math.pow(digit, 3);
            num /= 10;
        }

        if(cubeSum == originalNum)
            System.out.println(originalNum+ " is an Armstrong number");
        else
            System.out.println(originalNum+ " is not an Armstrong number");
    }
}